addons/l10n_id_efaktur/models/account_move.py

```diff
@@ -356,16 +355,51 @@ def _generate_efaktur(self, delimiter):
         my_utf8 = output_head.encode("utf-8")
         out = base64.b64encode(my_utf8)
 
-        attachment = self.env['ir.attachment'].create({
-            'datas': out,
-            'name': 'efaktur_%s.csv' % (fields.Datetime.to_string(fields.Datetime.now()).replace(" ", "_")),
-            'type': 'binary',
-        })
+        # can only generate efaktur doc IF:
+        # (1) no linked documents for all invoices; OR
+        # (2) all invoices are linked to same document AND document strictly links to these invoices as well;
+        attachment = self.env['ir.attachment']
+        if not self.l10n_id_efaktur_document:
+            self.l10n_id_efaktur_document = self.env['l10n_id_efaktur.document'].create({'invoice_ids': self.ids})
+        elif len(self.l10n_id_efaktur_document) == 1 and set(self.l10n_id_efaktur_document.invoice_ids.ids) == set(self.ids):
+            attachment = self.l10n_id_efaktur_document.attachment_id
+        else:
+            invoices_with_document = self.filtered(lambda x: x.l10n_id_efaktur_document).mapped('name')
+            raise UserError(_("Unable to generate efaktur document on invoies with pre-existing documents: %s", ','.join(invoices_with_document)))
+
+        # create/update attachment and link it to efaktur document if new
+        if not attachment:
+            attachment = self.env['ir.attachment'].create({
+                'datas': out,
+                'name': 'efaktur_%s.csv' % (fields.Datetime.to_string(fields.Datetime.now()).replace(" ", "_")),
+                'type': 'binary',
+                'res_model': 'l10n_id_efaktur.document',
+                'res_id': self.l10n_id_efaktur_document.id,
+            })
+            self.l10n_id_efaktur_document.attachment_id = attachment.id
+        else:
+            attachment.write({
+                'datas': out,
+                'name': 'efaktur_%s.csv' % (fields.Datetime.to_string(fields.Datetime.now()).replace(" ", "_")),
+            })
 
         for record in self:
             record.message_post(attachment_ids=[attachment.id])
```

Comment:
```suggestion
        for record in self:
            record.with_context(no_new_invoice=True).message_post(attachment_ids=[attachment.id])
```
To remove the "The invoice is not a draft, it was not updated from the attachment." and the logic linked to that

Reply:
Also adding a short message wouldn't be a bad idea :thinking: